lib/Raven/Client.php

```diff
@@ -462,76 +412,51 @@ public function capture(array $payload)
         }
 
         $event = $this->callMiddlewareStack($event, static::isHttpRequest() ? ServerRequestFactory::fromGlobals() : null, isset($payload['exception']) ? $payload['exception'] : null, $payload);
+        $event = $this->sanitize($event);
 
-        $payload = $event->toArray();
-
-        $this->sanitize($payload);
-
-        if (!$this->storeErrorsForBulkSend) {
-            $this->send($payload);
-        } else {
-            $this->pendingEvents[] = $payload;
-        }
+        $this->send($event);
 
         $this->lastEvent = $event;
 
-        return $payload['event_id'];
+        return str_replace('-', '', $event->getId()->toString());
     }
 
-    public function sanitize(&$data)
+    public function sanitize(Event $event)
     {
         // attempt to sanitize any user provided data
-        if (!empty($data['request'])) {
-            $data['request'] = $this->serializer->serialize($data['request']);
+        $request = $event->getRequest();
+        $userContext = $event->getUserContext();
+        $extraContext = $event->getExtraContext();
+        $tagsContext = $event->getTagsContext();
+
+        if (!empty($request)) {
+            $event = $event->withRequest($this->serializer->serialize($request));
         }
-        if (!empty($data['user'])) {
-            $data['user'] = $this->serializer->serialize($data['user'], 3);
+        if (!empty($userContext)) {
+            $event = $event->withUserContext($this->serializer->serialize($userContext, 3));
         }
-        if (!empty($data['extra'])) {
-            $data['extra'] = $this->serializer->serialize($data['extra']);
+        if (!empty($extraContext)) {
+            $event = $event->withExtraContext($this->serializer->serialize($extraContext));
         }
-        if (!empty($data['tags'])) {
-            foreach ($data['tags'] as $key => $value) {
-                $data['tags'][$key] = @(string) $value;
+        if (!empty($tagsContext)) {
+            foreach ($tagsContext as $key => $value) {
+                $tagsContext[$key] = @(string) $value;
```

Comment:
Why silencing errors? Maybe serializing those tags could be a better approach?

Reply:
Indeed, but I did not change the way serialization works. The code silencing the errors cames from version `1.x` and I have plans to modify the whole sanitize function anyway in a later PR, so I did not address the thing here